### PR TITLE
Add base error display alongside existing errors

### PR DESCRIPTION
**Pull request by bluefalcon26**
#1466

### CODE REVIEW

**Comment by fabriziobertoglio1987** [comment]
Thanks a lot. Sorry, I am a beginner.. Is there somewhere an article on how to include this in my project master branch? 

Thanks a lot
Fabrizio

**Comment by bluefalcon26** [comment]
@fabriziobertoglio1987 You should be able to do a git pull on 
https://github.com/bluefalcon26/simple_form.git:addBaseErrorDisplay

**Comment by fabriziobertoglio1987** [comment]
Ok, thanks a lot. I will search for instruction on github to do a pull. I never have done one .. 

Thanks 
Fabrizio

File: lib/generators/simple_form/templates/_form.html.erb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 <%%= simple_form_for(@<%= singular_table_name %>) do |f| %>
   <%%= f.error_notification %>
+  <%%= f.error_notification message: f.object.errors[:base].to_sentence if f.object.errors[:base].present? %>
```
Comment (feliperenan) [inline lib/generators/simple_form/templates/_form.html.erb]: You can use `f.error :base` instead.

**Comment by feliperenan** [comment]
@bluefalcon26 Thank you for the pull request 😄 . Could you rebase it with the master branch?

I don't see problems in adding it in the generator, @rafaelfranca any thoughts?

File: lib/generators/simple_form/templates/_form.html.erb (hunk 1 above)
Comment (bluefalcon26) [reply lib/generators/simple_form/templates/_form.html.erb]: Does f.error :base automatically check whether an error is actually present?
Don't want to generate extra markup for no error message.

IE, Should the line say 
`<%%= f.error :base %>`
or
`<%%= f.error :base if ... %>`

See here for more info:
https://github.com/plataformatec/simple_form/issues/1330

Comment (feliperenan) [reply lib/generators/simple_form/templates/_form.html.erb]: > Does f.error :base automatically check whether an error is actually present?

Yes, it is.

> Don't want to generate extra markup for no error message.

According to my tests, it doesn't generate extra markup. The only difference that I saw was that with `f.error` uses `span` tag with class `error` instead `div` with class `error_notification`.

```ruby
f.error_notification
=> <div class="error_notification">Custom validation</div>

f.error
=> <span class="error">Custom validation</span>
```

**Comment by collimarco** [comment]
Is this pull request the reason why there are two "duplicate" lines?

```
<%= f.error_notification %>
<%= f.error_notification message: f.object.errors[:base].to_sentence if f.object.errors[:base].present? %>
```

What is the reason for having this duplication?

**Comment by bluefalcon26** [comment]
> Is this pull request the reason why there are two "duplicate" lines?
> 
> ```
> <%= f.error_notification %>
> <%= f.error_notification message: f.object.errors[:base].to_sentence if f.object.errors[:base].present? %>
> ```
> 
> What is the reason for having this duplication?

f.error_notification <> f.object.errors[:base]

The second line adds error details if they are present. In your case, are you getting error details in the first line, or are you getting two error lines without details?

**Comment by collimarco** [comment]
@bluefalcon26 Thanks for the reply. Does `f.error_notification` alone display only a fixed message defined in YML (e.g. There are some errors) or it may include the specific runtime errors on some occasions? I'm tempted to remove the first `f.error_notification` since the errors are already displayed nicely below each field.

**Comment by bluefalcon26** [comment]
Go ahead, it's possible things have changed since I was working with this
and it may be superceded. Only thing I would say is before you do, double
check the interaction on nested forms. It's been a long time but I seem to
recall that being part of the rationale for my original pull request.

Daniel

**Pull request merged**